src/Auth/User/SamlUserFactory.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Kimai time-tracking app.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Auth\User;
+
+use App\Entity\User;
+use Hslavich\OneloginSamlBundle\Security\Authentication\Token\SamlTokenInterface;
+use Hslavich\OneloginSamlBundle\Security\User\SamlUserFactoryInterface;
+
+final class SamlUserFactory implements SamlUserFactoryInterface
+{
+    /**
+     * @var array
+     */
+    private $mapping;
+    /**
+     * @var string
+     */
+    private $groupAttribute;
+    /**
+     * @var array
+     */
+    private $groupMapping;
+
+    public function __construct(array $attributes)
+    {
+        $this->mapping = $attributes['mapping'];
+        $this->groupAttribute = $attributes['roles']['attribute'];
+        $this->groupMapping = $attributes['roles']['mapping'];
+    }
+
+    public function createUser(SamlTokenInterface $token)
+    {
+        $user = new User();
+        $user->setEnabled(true);
+        $user->setUsername($token->getUsername());
+
+        $this->hydrateUser($user, $token);
+
+        return $user;
+    }
+
+    public function hydrateUser(User $user, SamlTokenInterface $token): void
+    {
+        // extract user roles from a special saml attribute
+        if (!empty($this->groupAttribute) && $token->hasAttribute($this->groupAttribute)) {
+            $groupMap = [];
+            foreach ($this->groupMapping as $mapping) {
+                $field = $mapping['kimai'];
+                $attribute = $mapping['saml'];
+                $groupMap[$attribute] = $field;
```

Comment:
Not a new feature request but is there a simple way to concatenate firstname and lastname that some IdP provide.  Something like:

        mapping:
            - { saml: $firstname . ' ' . $lastname, kimai: alias }

This might work now, I have not checked.  I don't need it for my installation

Reply:
I thought about that as well, but I'll skip that for now as you don't need it ...